File: DS-Algo-with-Java/Lecture-13 Arrays/LastOccurrences.java
import java.util.Scanner;

public class LastOccurrences {

    static int Last(int arr[], int value) {

        int location = -1;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == value) {
                location = i;
            }
        }
        return location;
    }

    public static void main(String[] args) {
        
        try (Scanner sc = new Scanner(System.in)) {
            
            System.out.print("Enter size of array: ");
            int n = sc.nextInt();

            int arr[] = new int[n];
            System.out.println("Enter the elements: ");
            
            for (int i = 0; i < n; i++) {
                arr[i] = sc.nextInt();
            }

            System.out.print("Enter value for see last occurrence: ");
            int value = sc.nextInt();

            System.out.println("The value " + value + " last occurred at " + Last(arr, value) + " position in array.");
        }
    }
}

File: DS-Algo-with-Python/Lecture-13 Arrays/LastOccurrences.py
from array import *
    
def Last(arr, value):
    location = -1
    for i in range(len(arr)):
        if arr[i] == value:
            location = i
    return location

n = int(input("Enter size of array: "))

arr = array("i")
print("Enter the elements: ")
for i in range(n):
    arr.append(int(input()))

value = int(input("Enter value for see last occurrence: "))

print(f"The value {value} last occurred at {Last(arr, value)} position in array.")
